cli/src/commands/git/push.rs

```diff
@@ -968,25 +974,35 @@ fn find_bookmarks_targeted_by_revisions<'a>(
 ) -> Result<Vec<(&'a RefName, LocalAndRemoteRef<'a>)>, CommandError> {
     let mut revision_commit_ids = HashSet::new();
     if use_default_revset {
-        // remote_bookmarks(remote=<remote>)..@
-        let workspace_name = workspace_command.workspace_name();
-        let expression = RevsetExpression::remote_bookmarks(
-            StringPattern::everything(),
-            StringPattern::exact(remote),
-            None,
-        )
-        .range(&RevsetExpression::working_copy(workspace_name.to_owned()))
-        .intersection(&RevsetExpression::bookmarks(StringPattern::everything()));
+        // Get the default revset specified by the user.
+        let push_revset = workspace_command
+            .settings()
+            .get_string("revsets.git-push")?;
+        let mut context = workspace_command.env().revset_parse_context();
+        context.local_variables.insert(
+            "remote",
+            ExpressionNode {
+                kind: ExpressionKind::StringPattern {
+                    kind: "exact",
+                    value: remote.as_str().to_string(),
+                },
+                span: Span::new(remote.as_str(), 0, remote.as_str().len())
+                    .expect("programmatic span shouldn't fail"),
+            },
+        );
+        let mut diags = RevsetDiagnostics::default();
+        let (mut expression, _) = revset::parse_with_modifier(&mut diags, &push_revset, &context)?;
```

Comment:
nit: can you insert `let (None | Some(RevsetModifier::All)) = modifier;` to ensure that compilation would fail if we add new modifier?